bpf/cilium-probe-kernel-hz.c

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright (C) 2020 Authors of Cilium */
+
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <unistd.h>
+#include <sched.h>
+#include <errno.h>
+#include <string.h>
+#include <time.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <sys/resource.h>
+
+struct cpu_jiffies {
+	uint64_t *jiffies;
+	uint32_t cpus;
+};
+
+static const uint64_t kernel_hz[] = { 100, 250, 300, 1000 };
+
+#define abs(x)	({ x < 0 ? -x : x; })
+
+static int pin_to_cpu(int cpu)
+{
+	cpu_set_t set;
+
+	CPU_ZERO(&set);
+	CPU_SET(cpu, &set);
+
+	return sched_setaffinity(getpid(), sizeof(set), &set);
```

Comment:
Nit: could use `sched_setaffinity(0, ...)` here to avoid the getpid call (from http://man7.org/linux/man-pages/man2/sched_setaffinity.2.html: _If pid is zero, then the calling thread is used_). Not that it matters much, though.
```suggestion
	return sched_setaffinity(0, sizeof(set), &set);
```